pkg/server/batch_check.go

```diff
@@ -0,0 +1,130 @@
+package server
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openfga/openfga/pkg/middleware/validator"
+
+	"github.com/openfga/openfga/internal/condition"
+	"github.com/openfga/openfga/internal/graph"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/openfga/openfga/internal/authz"
+	"github.com/openfga/openfga/pkg/server/commands"
+	"github.com/openfga/openfga/pkg/telemetry"
+)
+
+func (s *Server) BatchCheck(ctx context.Context, req *openfgav1.BatchCheckRequest) (*openfgav1.BatchCheckResponse, error) {
+	ctx, span := tracer.Start(ctx, authz.BatchCheck, trace.WithAttributes(
+		attribute.KeyValue{Key: "store_id", Value: attribute.StringValue(req.GetStoreId())},
+		attribute.KeyValue{Key: "batch_size", Value: attribute.IntValue(len(req.GetChecks()))},
+		attribute.KeyValue{Key: "consistency", Value: attribute.StringValue(req.GetConsistency().String())},
+	))
+	defer span.End()
+
+	if !validator.RequestIsValidatedFromContext(ctx) {
+		if err := req.Validate(); err != nil {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+	}
+
+	err := s.checkAuthz(ctx, req.GetStoreId(), authz.BatchCheck)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx = telemetry.ContextWithRPCInfo(ctx, telemetry.RPCInfo{
+		Service: s.serviceName,
+		Method:  authz.BatchCheck,
+	})
+
+	storeID := req.GetStoreId()
+
+	typesys, err := s.resolveTypesystem(ctx, storeID, req.GetAuthorizationModelId())
+	if err != nil {
+		return nil, err
+	}
+
+	cmd := commands.NewBatchCheckCommand(
+		s.checkDatastore,
+		s.checkResolver,
+		typesys,
+		commands.WithBatchCheckCommandCacheController(s.cacheController),
+		commands.WithBatchCheckCommandLogger(s.logger),
+		commands.WithBatchCheckMaxChecksPerBatch(s.maxChecksPerBatchCheck),
+		commands.WithBatchCheckMaxConcurrentChecks(s.maxConcurrentChecksPerBatch),
+	)
+
+	result, err := cmd.Execute(ctx, &commands.BatchCheckCommandParams{
+		AuthorizationModelID: req.GetAuthorizationModelId(),
```

Comment:
```suggestion
		AuthorizationModelID: typesys.GetModelID()
```

please add a test for this... i think that if the user calls the API without a model ID we are finding the latest model (in line 51) but then the way you wrote it will pass an empty string to `cmd.Execute` and i'm not sure what that will do